MQ2AutoSize.cpp

```diff
@@ -331,13 +497,24 @@ PLUGIN_API void OnEndZone()
 
 PLUGIN_API void OnPulse()
 {
-	if (GetGameState() != GAMESTATE_INGAME || !AS_Config.OptByRange) return;
+	if (GetGameState() != GAMESTATE_INGAME)
+	{
+		return;
+	}
+
 	if (uiSkipPulse < SKIP_PULSES)
 	{
 		uiSkipPulse++;
 		return;
 	}
 
+	// check if communication plugins are still running and adjust UI as needed
+	if (GetTickCount64() > commsCheck)
+	{
+		commsCheck = std::int64_t(commsCheck) + (static_cast<long long>(commsRefreshRateSeconds) * 1000);
```

Comment:
not really sure what is going on here with the casts, but if you fix the types as i mentioned at the top of the file you shouldn't need these.